docs/source/blueprints/BP005.rst

```diff
@@ -0,0 +1,291 @@
+BP005
+#####
+
+:Number: BP005
+:Title: autils project plan: An unified standalone repository for utility libraries
+:Author: Beraldo Leal <bleal@redhat.com>, Cleber Rosa <crosa@redhat.com>
+:Discussions-To: avocado-devel@redhat.com or PR thread
+:Reviewers: Plamen Dimitrov <pdimitrov@pevogam.com>
+:Created: 14-June-2022
+:Type: Process Blueprint
+:Status: Draft
+
+.. contents:: Table of Contents
+
+TL;DR
+*****
+
+This blueprint describes the plan for migrating the utilities libraries present
+in Avocado and related projects to a standalone and central repository. Such a
+repository, called autils, is about having well tested utilities that will keep
+test writers efforts in the features to be tested, and away from debugging and
+troubleshooting those libraries. While this repository initially targets
+Avocado users writing tests, it should be generic enough for consumption by
+third-party developers outside of the Avocado community.  It presents a glimpse
+of the infrastructure that will guarantee quality criterias and will ease the
+onboarding and maintenance of such libraries.
+
+Besides the plan, with the migration steps for the current libraries, this
+blueprint also presents the tasks already known to be necessary in order to
+achieve this goal.
+
+
+.. note:: When reading here, in this document, "avocado.utils", please take
+          into consideration Avocado and Avocado-VT utility libraries.
+
+
+Motivation
```

Comment:
One of part I would like to propose to add is the clear definition scope of this project. I understand it may be hard at the beginning of project. Some practices learned from successful and health upstream project ,such as kubernetes, is that those projects  define very clear project scope definition, where it is clear to point out which part should be in , which part  is not good fit-in.